src/util/gui/menu.c

```diff
@@ -18,30 +18,26 @@ DEFINE_VECTOR(GUIMenuItemList, struct GUIMenuItem);
 DEFINE_VECTOR(GUIMenuSavedList, struct GUIMenuSavedState);
 
 void _itemNext(struct GUIMenuItem* item, bool wrap) {
-	if (item->state < item->nStates - 1) {
+	if (wrap || item->state < item->nStates - 1) {
 		unsigned oldState = item->state;
 		do {
-			++item->state;
+			item->state = (item->state + 1) % item->nStates;
```

Comment:
would a branch if state >= item->nStates be faster than a modulo operation? This isn't hot code, but I try to avoid div/mod wherever possible in mGBA, even in cold code.